src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -15,34 +16,87 @@ import {ICoExpressionPlotProps} from "./CoExpressionPlot";
+                    return this.props.store.coexpressionTabMolecularProfilesXGene.result[0];
         } else {
-            return this._selectedMolecularProfile;
+            return this._selectedProfileXGene;
         }
     }
 
-    @computed get selectedEntrezGeneId():number | undefined {
-        if (this._selectedEntrezGeneId === undefined && this.props.store.genes.isComplete &&
-                this.props.store.genes.result.length > 0) {
-            return this.props.store.genes.result[0].entrezGeneId;
+    @computed get selectedProfileXGeneset():MolecularProfile|undefined {
+        if (!this._selectedProfileXGeneset && this.props.store.coexpressionTabMolecularProfilesXGeneset.isComplete
+                && this.props.store.coexpressionTabMolecularProfilesXGeneset.result.length > 0) {
+                    return this.props.store.coexpressionTabMolecularProfilesXGeneset.result[0];
         } else {
-            return this._selectedEntrezGeneId;
+            return this._selectedProfileXGeneset;
+        }
+    }
+
+    @computed get selectedProfileY():MolecularProfile|undefined {
+        if (!this._selectedProfileY && this.props.store.coexpressionTabMolecularProfilesY.isComplete
+            && this.props.store.coexpressionTabMolecularProfilesY.result.length > 0) {
+            return this.props.store.coexpressionTabMolecularProfilesY.result[0];
+        } else {
+            return this._selectedProfileY;
+        }
+    }
+
+    @computed get selectedGeneticEntityId():string | undefined {
+        if (!this._selectedGeneticEntityId && this.props.store.geneticEntities !== undefined) {
+            return this.props.store.geneticEntities[0].geneticEntityId.toString();
+        } else {
+            return this._selectedGeneticEntityId;
+        }
+    }
+
+    @computed get isSelectedGeneticEntityAGeneSet(): boolean {
+        if (this._selectedGeneticEntityId !== undefined) {
+            for (const geneticEntity of this.props.store.geneticEntities!) {
+                if (geneticEntity.geneticEntityType === GeneticEntityType.GENESET && geneticEntity.geneticEntityId.toString() === this._selectedGeneticEntityId) {
+                    return true;
+                }
+            }
+            return false;
+        } else {
+            return this._isSelectedGeneticEntityAGeneSet;
+        }
+    }
+
+    @computed get selectedGeneticEntityName(): string | undefined {
+        if (this.props.store.geneticEntities !== undefined) {
+            if (this._selectedGeneticEntityId !== undefined) {
+                for (const geneticEntity of this.props.store.geneticEntities) {
+                    if (this._selectedGeneticEntityId === geneticEntity.geneticEntityId.toString()) {
+                        return geneticEntity.geneticEntityName;
+                    }
+                }
+                return this._selectedGeneticEntityName; //Expected not to reach this code, if so, it will return undefined
```

Comment:
if it's not set then why hold onto the variable just for it to be `undefined`? just return `undefined`